apps/sms/views/conversation/js/conversation.js

```diff
@@ -547,34 +562,39 @@ var ConversationView = {
   },
 
   beforeEnterThread: function conv_beforeEnterThread(args) {
-    // TODO should we implement hooks to Navigation so that Threads could
-    // get an event whenever the panel changes?
-    Threads.currentId = args.id;
+    var threadId = +args.id;
+    return MessageManager.ensureThreadRegistered(threadId).then(() => {
```

Comment:
Sorry I'm not quite sure why do we need to ensure the thread is registered or not before rendering, because we also register the message while rendering. Will is cause any problem if we don't do so?